Gems/ROS2/Code/Source/Gripper/VacuumGripper.cpp

```diff
@@ -0,0 +1,335 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root
+ * of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "VacuumGripper.h"
+#include "Utils.h"
+
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+#include <AzFramework/Components/TransformComponent.h>
+
+#include "Source/ArticulationLinkComponent.h"
+#include <AzFramework/Physics/PhysicsSystem.h>
+#include <AzFramework/Physics/RigidBodyBus.h>
+#include <AzFramework/Physics/SimulatedBodies/RigidBody.h>
+#include <imgui/imgui.h>
+
+namespace ROS2
+{
+    void VacuumGripper::Activate()
+    {
+        m_gripperEffectorBodyHandle = AzPhysics::InvalidSimulatedBodyHandle;
+        m_onTriggerEnterHandler = AzPhysics::SimulatedBodyEvents::OnTriggerEnter::Handler(
+            [&]([[maybe_unused]] AzPhysics::SimulatedBodyHandle bodyHandle, [[maybe_unused]] const AzPhysics::TriggerEvent& event)
+            {
+                AZ_Printf("ROS2", "VacuumGripper::Activate() - OnTriggerEnter\n");
+                const auto boxId = event.m_otherBody->GetEntityId();
+                const bool isGrippable = isObjectGrippable(boxId);
+                if (isGrippable)
+                {
+                    m_grippedObjectInEffector = boxId;
+                }
+            });
+
+        m_onTriggerExitHandler = AzPhysics::SimulatedBodyEvents::OnTriggerExit::Handler(
+            [&]([[maybe_unused]] AzPhysics::SimulatedBodyHandle bodyHandle, [[maybe_unused]] const AzPhysics::TriggerEvent& event)
+            {
+                AZ_Printf("ROS2", "VacuumGripper::Activate() - OnTriggerExit\n");
+                const auto boxId = event.m_otherBody->GetEntityId();
+                const bool isGrippable = isObjectGrippable(boxId);
```

Comment:
Wouldn't it be enough to check whether it is the entity equals m_grippedObjectInEffector (if not Invalid)?

Reply:
Nice catch!